class.jetpack-cli.php

```diff
@@ -1476,6 +1476,128 @@ private function partner_provision_error( $error ) {
+		$title = $args[0];
+
+		$slug = isset( $assoc_args['slug'] )
+			? $assoc_args['slug']
+			: sanitize_title( $title );
+
+		if ( ! preg_match( '/^[a-z][a-z0-9\-]*$/', $slug ) ) {
+			WP_CLI::error( esc_html__( 'Invalid block slug. They can contain only lowercase alphanumeric characters or dashes, and start with a letter', 'jetpack' ) . ' 👻' );
+		}
+
+		global $wp_filesystem;
+		if ( ! WP_Filesystem() ) {
+			WP_CLI::error( esc_html__( "Can't write files", 'jetpack' ) . ' 😱' );
+		}
+
+		$path = JETPACK__PLUGIN_DIR . "extensions/blocks/$slug";
+
+		if ( $wp_filesystem->exists( $path ) && $wp_filesystem->is_dir( $path ) ) {
+			WP_CLI::error( sprintf( esc_html__( 'Name conflicts with the existing block %s', 'jetpack' ), $path ) . ' ⛔️' );
+			exit( 1 );
+		}
+
+		$wp_filesystem->mkdir( $path );
+
+		$files = array(
+			"$path/$slug.js" => $this->render_block_file( 'block-register-php', array(
+				'slug' => $slug,
+				'title' => $title,
+			) ),
+			"$path/index.js" => $this->render_block_file( 'block-index-js', array(
+				'slug' => $slug,
+				'title' => $title,
+				'description' => isset( $assoc_args['description'] )
+					? $assoc_args['description']
+					: $title,
+				'keywords' => isset( $assoc_args['keywords'] )
+					? array_map( function( $keyword ) {
+						// Construction necessary for Mustache lists
+						return array( 'keyword' => trim( $keyword ) );
+					}, explode( ',', $assoc_args['keywords'] ) )
+					: '',
+				'externalEdit' => !! isset( $assoc_args['external-edit'] )
+			) ),
+			"$path/editor.js" => $this->render_block_file( 'block-editor-js' ),
+			"$path/editor.scss" => $this->render_block_file( 'block-editor-scss', array(
+				'title' => $title,
+			) ),
+		);
+
+		if ( isset( $assoc_args['external-edit'] ) ) {
+			$className = str_replace( ' ', '', ucwords( str_replace( '-', ' ', $slug ) ) );
+			$files[ "{$path}/edit.js" ] = $this->render_block_file( 'block-edit-js', array(
+				'title' => $title,
+				'className' => $className,
+			) );
+		}
+
+		$files_written = array();
+
+		foreach ( $files as $filename => $contents ) {
+			if ( $wp_filesystem->put_contents( $filename, $contents ) ) {
+				$files_written[] = $filename;
+			} else {
+				WP_CLI::error( sprintf( esc_html__( 'Error creating %s', 'jetpack' ), $filename ) );
+			}
+		}
+
+		if ( empty( $files_written ) ) {
+			WP_CLI::log( esc_html__( 'No files were created' ) );
+		} else {
+			// Load index.json and insert the slug of the new block in the production array
+			$file_block_list = JETPACK__PLUGIN_DIR . 'extensions/setup/index.json';
+			$block_list = $wp_filesystem->get_contents( $file_block_list );
+			if ( empty( $block_list ) ) {
+				WP_CLI::error( sprintf( esc_html__( 'Error fetching contents of %s', 'jetpack' ), $file_block_list ) );
+			} else if ( false === stripos( $block_list, $slug ) ) {
+				$new_block_list = str_replace( '"production": [', "\"production\": [\n\t\t\"$slug\",", $block_list );
```

Comment:
What do you think if we would place the block in `beta` list initially? It's the recommended way to start developing blocks.

Reply:
Done